src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -105,14 +105,15 @@ def __init__(self, dim, max_position_embeddings=2048, base=10000, device=None):
         self.max_position_embeddings = max_position_embeddings
         self.base = base
         self.register_buffer("inv_freq", None, persistent=False)
+        self.inv_freq = 1.0 / (
+            self.base ** (torch.arange(0, self.dim, 2, dtype=torch.int64).float() / self.dim)
+        )
 
     @torch.no_grad()
     def forward(self, x, position_ids, seq_len=None):
         # x: [bs, num_attention_heads, seq_len, head_size]
-        if self.inv_freq is None:
-            self.inv_freq = 1.0 / (
-                self.base ** (torch.arange(0, self.dim, 2, dtype=torch.int64, device=x.device).float() / self.dim)
-            )
+        self.inv_freq.to(device=x.device)
+
```

Comment:
This fix is necessary to make torch.compile work with this model. I will revert the change in this PR, but we should check what bets fix to do in another PR.